src/internal_events/kubernetes_logs.rs

```diff
@@ -8,7 +8,13 @@ use crate::event::Event;
 pub struct KubernetesLogsEventsReceived<'a> {
     pub file: &'a str,
     pub byte_size: usize,
-    pub pod_name: Option<&'a str>,
+    pub pod_info: Option<KubernetesLogsPodInfo>,
+}
+
+#[derive(Debug)]
+pub struct KubernetesLogsPodInfo {
+    pub name: String,
+    pub namespace: String,
 }
```

Comment:
Shouldn't we keep these as references, to keep the metric details as cheap as possible?

I would've expected something like:

```rust

pub struct KubernetesLogsPodInfo<'a> {
    pub name: &'a str,
    pub namespace: &'a str,
}
```

Reply:
@jszwedko (blaming you because you typed this) should I make this change?

Reply:
I think we ran into issue using references there, but I can't remember what they were. The data needs to be owned when passing it into the metrics and trace macros anyway so we probably wouldn't save that much, but I'd support refactoring it.

Reply:
```
error[E0515]: cannot return value referencing temporary value
   --> src/sources/kubernetes_logs/mod.rs:419:57
    |
419 |                   pod_info: file_info.as_ref().map(|info| KubernetesLogsPodInfo {
    |  _________________________________________________________^
420 | |                     name: &info.pod_name.to_owned(),
421 | |                     namespace: &info.pod_namespace.to_owned(),
    | |                                 ----------------------------- temporary value created here
422 | |                 }),
    | |_________________^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing temporary value
   --> src/sources/kubernetes_logs/mod.rs:419:57
    |
419 |                   pod_info: file_info.as_ref().map(|info| KubernetesLogsPodInfo {
    |  _________________________________________________________^
420 | |                     name: &info.pod_name.to_owned(),
    | |                            ------------------------ temporary value created here
421 | |                     namespace: &info.pod_namespace.to_owned(),
422 | |                 }),
    | |_________________^ returns a value referencing data owned by the current function

error[E0495]: cannot infer an appropriate lifetime due to conflicting requirements
  --> src/internal_events/kubernetes_logs.rs:28:15
   |
28 |         match self.pod_info {
   |               ^^^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime `'_` as defined here...
  --> src/internal_events/kubernetes_logs.rs:20:53
   |
20 | impl InternalEvent for KubernetesLogsEventsReceived<'_> {
   |                                                     ^^
note: ...so that the types are compatible
  --> src/internal_events/kubernetes_logs.rs:28:15
   |
28 |         match self.pod_info {
   |               ^^^^^^^^^^^^^
   = note: expected `std::option::Option<internal_events::kubernetes_logs::KubernetesLogsPodInfo<'_>>`
              found `std::option::Option<internal_events::kubernetes_logs::KubernetesLogsPodInfo<'_>>`
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the types are compatible
  --> src/internal_events/kubernetes_logs.rs:33:17
   |
33 |                 counter!("component_received_events_total", 1, "pod_name" => pod_name.clone(), "pod_namespace" => pod_namespace.clone());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected `<metrics::cow::Cow<'static, str> as From<&str>>`
              found `<metrics::cow::Cow<'_, str> as From<&str>>`
   = note: this error originates in the macro `counter` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0495, E0515.
For more information about an error, try `rustc --explain E0495`.
```